src/woo_publications/publications/admin.py

```diff
@@ -637,28 +640,68 @@ def get_changeform_initial_data(self, request: HttpRequest):
         initial_data["eigenaar"] = owner
         return initial_data
 
+    def get_readonly_fields(self, request: HttpRequest, obj=None):
+        readonly_fields = super().get_readonly_fields(request, obj)
+
+        if not obj:
+            readonly_fields += ("publicatiestatus",)
+
+        return readonly_fields
+
+    def has_change_permission(self, request, obj=None):
+        if obj and obj.publicatiestatus == PublicationStatusOptions.revoked:
+            return False
+        return super().has_change_permission(request, obj)
+
     def save_model(
         self, request: HttpRequest, obj: Document, form: forms.Form, change: bool
     ):
-        super().save_model(request, obj, form, change)
+        # set the new status to revoked if that is provided by the form
+        # otherwise use the parents publicatiestatus
+        new_status = (
+            PublicationStatusOptions.revoked
+            if form.cleaned_data.get("publicatiestatus")
+            == PublicationStatusOptions.revoked
+            else form.cleaned_data["publicatie"].publicatiestatus
+        )
 
-        new_status = obj.publicatiestatus
+        original_status = form.initial.get("publicatiestatus")
         is_published = new_status == PublicationStatusOptions.published
+        is_revoked = new_status == PublicationStatusOptions.revoked
 
-        if change:
-            original_status = form.initial["publicatiestatus"]
-            if new_status != original_status and not is_published:
+        same_publication = obj.publicatie.pk == form.initial.get("publicatie")
+        use_state_manager = (
+            obj.pk and same_publication and new_status != original_status
+        )
+
+        if use_state_manager:
+            obj.publicatiestatus = original_status or ""
+            match new_status:
+                case PublicationStatusOptions.concept:
```

Comment:
I think this is better done in the `Form.save` method (which is called by the django admin in `ModelAdmin.save_form`)